src/core_plugins/timelion/public/directives/timelion_expression_input_helpers.js

```diff
@@ -93,19 +93,26 @@ function extractSuggestionsFromParsedResult(result, cursorPosition, functionList
   // location range includes '.', function name, and '('.
   const openParen = activeFunc.location.min + activeFunc.function.length + 2;
   if (cursorPosition < openParen) {
-    return { list: [funcDefinition], location: activeFunc.location, type: SUGGESTION_TYPE.FUNCTIONS };
+    return { list: [functionHelp], location: activeFunc.location, type: SUGGESTION_TYPE.FUNCTIONS };
   }
 
   // Do not provide 'inputSeries' as argument suggestion for chainable functions
-  const args = funcDefinition.chainable ? funcDefinition.args.slice(1) : funcDefinition.args.slice(0);
+  const args = functionHelp.chainable ? functionHelp.args.slice(1) : functionHelp.args.slice(0);
 
   const activeArg = activeFunc.arguments.find((argument) => {
     return inLocation(cursorPosition, argument.location);
   });
   // return argument_value suggestions when cursor is inside agrument value
   if (activeArg && activeArg.type === 'namedArg' && inLocation(cursorPosition, activeArg.value.location)) {
-    // TODO - provide argument value suggestions once function list contains required data
-    return { list: [], location: activeArg.value.location, type: SUGGESTION_TYPE.ARGUMENT_VALUE };
+    const valueSuggestions = await getArgValueSuggestions(
```

Comment:
This would become:

```javascript
    const {
      function: functionName,
      arguments: functionArgs,
    } = activeFunc;

    const {
      name: argName,
      value: { text: partialInput },
    } = activeArg;

    let valueSuggestions;

    if (hasDynamicSuggestionsForArgument(functionName, argName)) {
      valueSuggestions = await getDynamicSuggestionsForArgument(functionName, argName, functionArgs, partialInput);
    } else {
      const {
        suggestions: staticSuggestions,
      } = functionHelp.args.find((arg) => {
        return arg.name === activeArg.name;
      });

      valueSuggestions = getStaticSuggestionsForInput(partialInput, staticSuggestions);
    }
```

Reply:
This comment pertains to my last comment.